iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformersTest.java

```diff
@@ -0,0 +1,444 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.commons.test.TestUtil;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ThreeSides;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+import com.powsybl.iidm.network.test.ThreeWindingsTransformerNetworkFactory;
+import com.powsybl.iidm.network.util.BranchData;
+import com.powsybl.iidm.network.util.TwtData;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import static com.powsybl.iidm.modification.TransformersTestUtils.*;
+import static com.powsybl.iidm.modification.TransformersTestUtils.addPhaseTapChanger;
+import static com.powsybl.iidm.modification.util.ModificationReports.lostTwoWindingsTransformerExtensions;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Luma Zamarreño {@literal <zamarrenolm at aia.es>}
+ * @author José Antonio Marqués {@literal <marquesja at aia.es>}
+ */
+class Replace3TwoWindingsTransformersByThreeWindingsTransformersTest {
+    private Network network;
+    private TwoWindingsTransformer t2w1;
+    private TwoWindingsTransformer t2w2;
+    private TwoWindingsTransformer t2w3;
+
+    @BeforeEach
+    public void setUp() {
+        Network expectedNetwork = createSetUpNetwork();
+        assertEquals(3, expectedNetwork.getTwoWindingsTransformerCount());
```

Comment:
Setup should not have asserts, just the creation of the objects needed to make the tests. if the objects created must be tested, define a simple test method below the setup with the checks.

Reply:
Done